src/lib/aws/construct/static-asset-deployment/main.ts

```diff
@@ -14,52 +17,117 @@ import { StaticAssetDeploymentProps } from './types'
  * class CustomConstruct extends StaticAssetDeployment {
  *   constructor(parent: Construct, id: string, props: StaticAssetDeploymentProps) {
  *     super(parent, id, props)
- *     this.props = props
- *     this.id = id
  *     this.initResources()
  *   }
  * }
  */
-export class StaticAssetDeployment extends CommonConstruct {
-  /* construct properties */
-  props: StaticAssetDeploymentProps
-  id: string
```

Comment:
the `id` value is on the `node` which you can just access so no need to store it again